R/dbi-methods.R

```diff
@@ -167,8 +170,13 @@ setMethod("dbSendQuery", c("SQLServerConnection", "character"),
     if (is_parameterised(ps) && is.null(params)) {
       return(pre_res)
     } else {
-      if (is_parameterised(ps) && !is.null(params)) {
-        dbBind(pre_res, params)
+      qry_nparams <- num_parameters(ps)
+      nparams <- length(params)
+      if (qry_nparams > 0 && nparams > 0L) {
+        if (nparams > qry_nparams) {
+          params <- params[1:qry_nparams]
+        }
+        dbBind(pre_res, params, batch = batch)
```

Comment:
If the number of parameters in the query is different than the length of the list, too-many or too-few bindings throws an error (perhaps obvious); it was either use this code or do bounds-checking/limiting within `dbBind`. Perhaps it's better to do the length-checking within `dbBind` instead of within the calling functions?